File: Mobile/lib/features/catalog/domain/entities/review.dart
import 'package:equatable/equatable.dart';

class Review extends Equatable {
  const Review({
    required this.id,
    required this.productId,
    required this.userName,
    required this.rating,
    required this.comment,
    required this.createdAt,
  });

  final String id;
  final String productId;
  final String userName;
  final int rating;
  final String comment;
  final DateTime createdAt;

  @override
  List<Object?> get props =>
      [id, productId, userName, rating, comment, createdAt];
}

File: API/ECommerceBackend/ECommerce.Domain/Catalog/Review.cs
using ECommerce.Domain.Common;
using ECommerce.Domain.Identity;

namespace ECommerce.Domain.Catalog;

public sealed class Review : EntityBase
{
    public string ProductId { get; set; } = string.Empty;
    public Product Product { get; set; } = null!;
    public string CustomerAccountId { get; set; } = string.Empty;
    public Account CustomerAccount { get; set; } = null!;
    public int Rating { get; set; }
    public string Comment { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;
    public DateTime? DeletedAtUtc { get; set; }
    public ICollection<ReviewPhoto> Photos { get; set; } = new List<ReviewPhoto>();
}
